docs/en/react/react-lynx-testing-library/index.mdx

```diff
@@ -0,0 +1,624 @@
+    >
+      <list-item
+        item-key="1"
+      >
+        <text>
+          1
+        </text>
+      </list-item>
+    </list>
+  `);
+});
+```
+
+### Main thread script
+
+> For the usage document of the main thread script, please refer to: https://lynxjs.org/zh/react/main-thread-script.html
+
+```jsx
+import { fireEvent, render } from '@lynx-js/react/testing-library';
+import { expect } from 'vitest';
+
+it('main thread script', async () => {
+  globalThis.cb = vi.fn();
+  const Comp = () => {
+    return (
+      <view
+        main-thread:bindtap={(e) => {
+          'main thread';
+          globalThis.cb(e);
+        }}
+      >
+        <text>Hello Main Thread Script</text>
+      </view>
+    );
+  };
+  const { container } = render(<Comp />, {
+    // You can try to enable both the main thread and the background thread at the same time
+    // and the results will be the same
+    // enableMainThread: true,
+    // enableBackgroundThread: true,
+  });
+  expect(container).toMatchInlineSnapshot(`
+    <page>
+      <view>
+        <text>
+          Hello Main Thread Script
+        </text>
+      </view>
+    </page>
+  `);
+  fireEvent.tap(container.firstChild, {
+    key: 'value',
+  });
+  expect(cb).toBeCalledTimes(1);
+  expect(cb.mock.calls).toMatchInlineSnapshot(`
+    [
+      [
+        {
+          "eventName": "tap",
+          "eventType": "bindEvent",
+          "isTrusted": false,
+          "key": "value",
+        },
+      ],
+    ]
+  `);
+});
+```
+
+#### Choose the thread used for rendering
+
+The second parameter of the `render` method supports `enableMainThread` and `enableBackgroundThread` are two options for enabling the main thread and background threads. By default, `enableMainThread` is `false` and `enableBackgroundThread` is `true`, so the main thread does not render the first screen, only the background thread runs the full Preact runtime and sends UI updates to the main thread for rendering. The reason why we don't enable both the main thread and the background thread is because unit tests do not compile files into two products like using Rspeedy does, which will result in the top-level code being executed only once, and by default in the background thread. Suppose you wrote a component like this:
+
+```jsx
+const isBackground = __BACKGROUND__;
+const Comp = () => {
+  return (
+    <view>
+      <text>{isBackground ? 'background' : 'main thread'}</text>
+    </view>
+  );
+};
+```
+
+The top-level `isBackground` of this component will only be executed once, and by default it is `true` (Lynx Environment defaults to the background thread after initialization). In order to avoid rendering the wrong first screen result when `enableMainThread: true`, we set `enableMainThread` to `false` by default, and get an empty first screen, which can avoid this problem.
```

Comment:
nit:
```suggestion
The top-level `isBackground` of this module will only be executed once, and by default it is `true` (Lynx Environment defaults to the background thread after initialization). In order to avoid rendering the wrong first screen result when `enableMainThread: true`, we set `enableMainThread` to `false` by default, and get an empty first screen, which can avoid this problem.
```